NetUtils.py

```diff
@@ -11,12 +11,87 @@
 from Utils import ByValue, Version
 
 
-class HintStatus(ByValue, enum.IntEnum):
-    HINT_UNSPECIFIED = 0
-    HINT_NO_PRIORITY = 10
-    HINT_AVOID = 20
-    HINT_PRIORITY = 30
-    HINT_FOUND = 40
+class HintStatus(ByValue, enum.IntFlag):
+    # Lower 5 bits: Priorities as integers
+    HINT_PRIORITY_UNSPECIFIED = 0  # For readable code
+    HINT_PRIORITY_NO_PRIORITY = 1
+    HINT_PRIORITY_AVOID = 2
+    HINT_PRIORITY_PRIORITY = 3
+    PRIORITY_MASK = 0b11111
+
+    # Bits 6+: Technical status
+    HINT_FOUND = 0b10000000
+
+    # For "compatibility", only used in CommonClient, thus should not affect memory for MultiServer
+    HINT_UNFOUND_LEGACY = OLD_HINT_FORMAT = 0b1000000000000000
+
+    @property
+    def priority(self):
```

Comment:
You might want to add the return types here, same for the other methods.

```suggestion
    def priority(self) -> HintStatus:
```

Reply:
Can you do that? I thought this causes a problem.

Reply:
Yeah, I guess you're right. Looking at `IntFlag`, the implementation of `&` does return `Self`, we should do the same here.

See https://docs.python.org/3/library/typing.html#typing.Self

Reply:
Well, that feature was introduced in Python 3.11, and we still support 3.10 :D

Reply:
We have `typing_extensions` tho, so you can use `typing_extensions.Self`

Reply:
That's true. I wonder if there are other instances of this in this file, if there are I feel like that should maybe be its own PR, if not then I can add it for sure

Reply:
Typing in this file seems not exactly consistent. Some methods are typed correctly, some are missing return types, some are missing args... I would not hold on adding typing to the methods of Hint/HintStatus because of that since you already changed those lines, but that's your call.